app/src/main/java/net/bible/service/download/Repositories.kt

```diff
@@ -53,83 +59,23 @@ abstract class RepoBase {
     }
 }
 
-class AndBibleRepo : RepoBase() {
-    override fun getRepoBooks(refresh: Boolean): List<Book> {
-        val bookList = getBookList(SUPPORTED_DOCUMENTS, refresh)
-        storeRepoNameInMetaData(bookList)
-        return bookList
-    }
-
-    override val repoName: String get() = REPOSITORY
-
-    companion object {
-        private const val REPOSITORY = "AndBible"
-        private val SUPPORTED_DOCUMENTS: BookFilter = AcceptableBookTypeFilter()
-    }
-}
-
-class StepRepo : RepoBase() {
-    override fun getRepoBooks(refresh: Boolean): List<Book> {
-        val bookList = getBookList(SUPPORTED_DOCUMENTS, refresh)
-        storeRepoNameInMetaData(bookList)
-        return bookList
-    }
-
-    override val repoName: String get() = REPOSITORY
-
-    companion object {
-        private const val REPOSITORY = "STEP Bible (Tyndale)"
-        private val SUPPORTED_DOCUMENTS: BookFilter = AcceptableBookTypeFilter()
-    }
-}
-
-class AndBibleExtraRepo : RepoBase() {
-    override fun getRepoBooks(refresh: Boolean): List<Book> {
-        val bookList = getBookList(SUPPORTED_DOCUMENTS, refresh)
-        storeRepoNameInMetaData(bookList)
-        return bookList
-    }
-
-    override val repoName: String get() = REPOSITORY
-
-    companion object {
-        private const val REPOSITORY = "AndBible Extra"
-        private val SUPPORTED_DOCUMENTS: BookFilter = AcceptableBookTypeFilter()
-    }
-}
+fun AndBibleRepo() = Repository("AndBible", AcceptableBookTypeFilter())
```

Comment:
Looks good! I'm not sure if we need to keep these as factory functions, or if one instance is enough. Merging and checking if it can be changed...